akka-actor/src/main/scala/akka/io/dns/DnsSettings.scala

```diff
@@ -0,0 +1,162 @@
+import scala.concurrent.duration.FiniteDuration
+import scala.io.Source
+import scala.util.{ Failure, Success, Try }
+import java.lang.reflect.Method
+import java.security.{ AccessController, PrivilegedExceptionAction }
+import java.util
+
+import scala.util.control.NonFatal
+
+/** INTERNAL API */
+private[dns] final class DnsSettings(c: Config) {
+  import DnsSettings._
+
+  val systemNameServers =
+    if (c.getBoolean("resolv-conf"))
+      parseSystemNameServers(Paths.get("/etc/resolv.conf").toFile)
+    else
+      Option.empty[immutable.Seq[InetSocketAddress]]
+
+  val NameServers: immutable.Seq[InetSocketAddress] = {
+    val addrs = Try(c.getString("nameservers")).toOption.toList
+      .flatMap {
+        case "default" ⇒ getDefaultSearchDomains().getOrElse(failUnableToDetermineDefaultNameservers)
+        case address   ⇒ parseNameserverAddress(address) :: Nil
+      }
+    if (addrs.nonEmpty) addrs
+    else c.getStringList("nameservers").asScala.map(parseNameserverAddress)(breakOut)
+  }
+
+  val NegativeTtl: Long = c.getDuration("negative-ttl", TimeUnit.MILLISECONDS)
+  val MinPositiveTtl: Long = c.getDuration("min-positive-ttl", TimeUnit.MILLISECONDS)
+  val MaxPositiveTtl: Long = c.getDuration("max-positive-ttl", TimeUnit.MILLISECONDS)
+
+  val ResolveTimeout: FiniteDuration = FiniteDuration(c.getDuration("request-ttl", TimeUnit.MILLISECONDS), TimeUnit.MILLISECONDS)
+
+  // -------------------------
+
+  private val inetSocketAddress = """(.*?)(?::(\d+))?""".r
+
+  def parseNameserverAddress(str: String): InetSocketAddress = {
+    val inetSocketAddress(host, port) = str
+    new InetSocketAddress(host, Option(port).fold(53)(_.toInt))
+  }
+
+  // TODO replace with actual parser, regex is very likely not efficient...
+  private val ipv4Address = """^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$""".r
+  private val ipv6Address = """^\s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|(([0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5})|((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:))|(:(((:[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)(\.(25[0-5]|2[0-4]\d|1\d\d|[1-9]?\d)){3}))|:)))(%.+)?\s*$""".r
```

Comment:
would best be replaced by a proper parboiled parser... but for now copied form akka-dns

Reply:
but dependency wise we can't do parboiled2 here, so we'll see...